util/math/math.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package math
+
+import "math"
+
+// Abs implement the abs function according to http://cavaliercoder.com/blog/optimized-abs-for-int64-in-go.html
+func Abs(n int64) int64 {
+	y := n >> 63
+	return (n ^ y) - y
+}
+
+// uintSizeTable is used as a table to do comparison to get uint length is faster than doing loop on division with 10
+var uintSizeTable = [21]uint64{
+	0, // redundant 0 here, so to make function StrLenOfUint64Fast to count from 1 and return i directly
+	9, 99, 999, 9999, 99999,
+	999999, 9999999, 99999999, 999999999, 9999999999,
+	99999999999, 999999999999, 9999999999999, 99999999999999, 999999999999999,
+	9999999999999999, 99999999999999999, 999999999999999999, 9999999999999999999,
+	math.MaxUint64,
+} // math.MaxUint64 is 18446744073709551615 and it has 20 digits
+
+// StrLenOfUint64Fast efficiently calculate the string character lengths of an uint64 as input
+func StrLenOfUint64Fast(x uint64) int {
+	for i := 1; ; i++ {
+		if x <= uintSizeTable[i] {
+			return i
+		}
+	}
+}
+
+// StrLenOfInt64Fast efficiently calculate the string character lengths of an int64 as input
+func StrLenOfInt64Fast(x int64) int {
```

Comment:
Seems that they are not used?

Reply:
Yes, just copied the whole file

Reply:
is this function introduced by another pr, and that pr is not cherry-picked to release 2.1?